### ISSUE TITLE
Templater resets active document position, if it needs to render a document in the background

### ISSUE THREAD

**Issue by stoilis** [SilentVoid13/Templater#1310]
TLDR; Bug #1297 I reported was supposed to be fixed after the latest release, but it looks like it's not. 
I've added a comment with a reproduction using the latest version [here](https://github.com/SilentVoid13/Templater/issues/1297#issuecomment-1906845936) but it doesn't look like it's visible after the bug was closed. So I am creating this new bug, to ensure it's noticed.

Tagging @Zachatoo too.

**Plugin information:**
OS: MacOS Sonoma 14.2.1
Templater version: 2.1.2
Obsidian version: 1.5.3
Templater settings: Folder Templates

**Describe the bug:**
When all of the following are true:
- I enable the use of Folder Templates
- I have a template associated to a specific folder
- Some outside process creates a blank file on that folder(which means Templater needs to render it)
then the following happens: The cursor in the current active document jumps to the top of the document. If I am currently writing something, the cursors unexpectedly "resets" to the top.
Note: The active document is just any other document in the vault. It's not related to either the template, the folder or the newly created document.

**Expected behavior**
I would expect the cursor to remain in its current position

**Screenshots**
In this video I start by showing the version of both Obsidian and Templater.
Then I initiate a command line to create a file after 5 seconds and then I move to start writing on a different document.
Notice how when the file created, the focus of my cursor moves to the top of the page.
![output2](https://github.com/SilentVoid13/Templater/assets/873390/39492691-7ff5-42af-a7d2-b96e4809874b)
https://github.com/SilentVoid13/Templater/assets/873390/5647fe41-a726-4d51-a163-a91c4725d889

**Additional context**
This bug may not look like much and I'll be the first to admit it's an edge case. But the automatic document creation I am using runs every 1 minute, making this issue particularly annoying.
I'd be curious to see if there is any better way of running Templater in the background that doesn't trigger this issue.

**Comment by Zachatoo** [comment]
Interesting, this time I can't reproduce. Can you also reproduce in the help vault with just Templater installed? Or any other details you can provide to aid me in reproducing?

https://github.com/SilentVoid13/Templater/assets/6936914/f3a0afd3-0419-443a-a828-c507eca17133

**Comment by stoilis** [comment]
After some trial-and-error, it seems to be reproducible only when I have [Dataview](https://github.com/blacksmithgu/obsidian-dataview) installed and enabled, regardless of whether the new file uses any Dataview-rendered blocks.

**Comment by stoilis** [comment]
Here is a video reproduction of a completely empty vault, almost all core plugins disabled, and only the latest Templater enabled.

https://github.com/SilentVoid13/Templater/assets/873390/51d4421a-1a06-4702-b60f-04f844c8fba3

You mentioned something about a "help vault". Not sure what that is, but if you point me to it, I'll give it a try there too.

**Comment by stoilis** [comment]
No changes in any setting, other than setting the general templates folder, and assigning the templates as the default, for the folder where I create the new file.
<img width="981" alt="Screenshot 2024-02-02 at 15 07 29" src="https://github.com/SilentVoid13/Templater/assets/873390/ad780acb-4c3c-4841-b14e-14837dbd7560">
<img width="963" alt="Screenshot 2024-02-02 at 15 07 07" src="https://github.com/SilentVoid13/Templater/assets/873390/26e15d41-b86f-43a4-bd7c-d042ba616e3d">

**Comment by Zachatoo** [comment]
Not sure what I was doing before, but I was able to reproduce. Should be actually fixed in 2.2.0, let me know if it doesn't work for you!

**Comment by stoilis** [comment]
Tested on 2.2.1 and seems to work fine. Thank you!
